homeassistant/components/srp_energy/__init__.py

```diff
@@ -0,0 +1,93 @@
+"""The SRP Energy integration."""
+import logging
+from srpenergy.client import SrpEnergyClient
+
+import voluptuous as vol
+
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME, CONF_ID
+import homeassistant.helpers.config_validation as cv
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN, DEFAULT_NAME
+
+_LOGGER = logging.getLogger(__name__)
+
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_ID): cv.string,
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up SRP Energy from a config entry."""
+    if DOMAIN in config:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+            )
+        )
```

Comment:
There is a move away from yaml configuration for integrations towards services, new integrations should not provide that interface.

Reply:
Hi @Kane610 do you have an example on a device that does the config properly?

Reply:
You should just remove this